contracts/blender/src/contract.rs

```diff
@@ -160,6 +168,34 @@ mod blender {
             self.current_root()
         }
 
+        /// Retrieve the path from the leaf to the root. `None` if the leaf does not exist.
+        #[ink(message, selector = 4)]
+        pub fn merkle_path(&self, leaf_idx: u32) -> Option<MerklePath> {
+            if self.max_leaves > leaf_idx || leaf_idx >= self.next_free_leaf {
```

Comment:
`self.max_leaves > leaf_idx` shouldn't be necessary as the second condition already covers it, right?

Reply:
nope, we want to ensure, that `leaf_idx` belongs to the interval `[self.max_leaves; self.next_free_leaf)`, so we have to check both boundaries

Reply:
Ah, of course. `leaf_idx` has to be less than `self.max_leaves` b/c at the position `0..self.max_leaves` we have intermediate nodes of the trie.

The smaller value (`leaf_idx`) being on the right side threw me off. My bad.